iceoryx_binding_c/source/c_runtime.cpp

```diff
@@ -23,13 +23,31 @@ extern "C" {
 
 void iox_runtime_register(const char* const name)
 {
-    PoshRuntime::getInstance(name);
+    if (name == nullptr)
+    {
+        LogError() << "Application name is a nullptr!";
+        std::terminate();
```

Comment:
Is it not a little bit to hard to terminate the complete process when the given name does not match the requirements? Maybe it should just print a message like "application name not set, runtime not created" and exit the function without creating a runtime.
Same for the process name length.

Reply:
Maybe we return a `bool` if it was successful. Or an enum which states the problem like:
- `APPLICATION_NAME_IS_NULLPTR`
- `EXCEEDING_ALLOWED_NAME_LENGTH`

Reply:
it's the same behavior like in the PoshRuntime. I also don't consider this a place for a graceful degradation. We need a runtime. Iceoryx doesn't work without a runtime. It's mandatory and nothing which only might be called under some conditions. It's essentially the same as when you don't get memory. Your application basically doesn't work when this call fails, so immediate termination makes sense here.

Reply:
@dkroenke since this is not a behavioral change (see PoshRuntime) are you okay with keeping it as is?

Reply:
yes it's fine